### PR TITLE
Feature/rollup refactor

### PR DESCRIPTION
**Pull request by MatthewCMoya**
## Description
* Opt out of rollup CLI and use JS api. Build time < 5 seconds
* Add sizes.json to umd
* Remove bash script for bundling

This gives us more efficient builds by not copying files for every single configuration and allowing concurrent bundling

## PR Requirements
Before requesting review this criteria must be met: 
Overall test coverage >= current master?
- [ ] Yes
- [x] N.A.

Documentation included (if any behavioral changes)?
- [ ] Yes
- [x] N.A.

Backwards compatibility (if breaking change)?
- [ ] Yes
- [x] N.A.

## Release
Will this pr trigger a release i.e. `version` in `package.json` has been bumped?
- [ ] Yes
- [x] No

### CODE REVIEW

File: rollup.build.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+async function copyFiles() {
+  return Promise.all(FILES_TO_COPY.map(async (file) => {
+    await fs.copy(file.path, `${BUILD_DIR}/${file.name}`)
+  }))
+}
+
+const startTime = new Date().getTime();
+createBundle()
+  .then(() => fs.writeJson(`${BUILD_DIR}/umd/sizes.json`, fileSizesObject))
+  .then(() => copyFiles())
```
Comment (stephenbaldwin) [inline rollup.build.js]: can you add the "create bundle code" as well?

Comment (stephenbaldwin) [reply rollup.build.js]: for bundle.production.min

File: rollup.build.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    const prod = filePath.endsWith('production.min.js');
+
+    if (dev || prod) {
+      const data = await fsRead(filePath, 'utf8');
+      dev ? development += data : production += data;
+    }
+  }
+
+  await fsWrite(`${UMD_DIR}/bundle.development.js`, development);
+  await fsWrite(`${UMD_DIR}/bundle.production.min.js`, production);
```
Comment (stephenbaldwin) [inline rollup.build.js]: could we promise.all these two and await that?

Comment (MatthewCMoya) [reply rollup.build.js]: Absolutely. Made a note to address in future pr along with multi-process rollup

**Pull request merged**